libs/repositories/src/repos/TokenBalancesRepository/TokenBalancesRepositoryMoralis.ts

```diff
@@ -0,0 +1,101 @@
+import { injectable } from 'inversify';
+import {
+  TokenBalanceParams,
+  TokenBalancesRepository,
+  TokenBalancesResponse,
+} from './TokenBalancesRepository';
+import {
+  MORALIS_API_BASE_URL,
+  MORALIS_API_KEY,
+  MORALIS_CLIENT_NETWORK_MAPPING,
+} from '../../datasources/moralis';
+
+type MoralisBalanceTokenResponse = {
+  token_address: string;
+  symbol: string;
+  name: string;
+  logo: string;
+  thumbnail: string;
+  decimals: number;
+  balance: string;
+  possible_spam: boolean;
+  verified_contract: boolean;
+  total_supply: string;
+  total_supply_formatted: string;
+  percentage_relative_to_total_supply: number;
+  security_score: number;
+};
+
+type MoralisBalanceResponse = {
+  result: MoralisBalanceTokenResponse[];
+};
+
+function isMoralisBalanceResponse(
+  data: unknown
+): data is MoralisBalanceResponse {
+  if (!data || typeof data !== 'object') {
+    return false;
+  }
+
+  const response = data as MoralisBalanceResponse;
+
+  if (!Array.isArray(response.result)) {
+    return false;
+  }
+
+  return !!response.result;
+}
+
+@injectable()
+export class TokenBalancesRepositoryMoralis implements TokenBalancesRepository {
+  async getTokenBalances({
+    address,
+    chainId,
+  }: TokenBalanceParams): Promise<TokenBalancesResponse> {
+    const network = MORALIS_CLIENT_NETWORK_MAPPING[chainId];
+    if (!network) {
+      throw new Error('Unsupported chain');
+    }
+
+    const response = await this.requestBalanceData(address, network);
+
+    if (!response.ok) {
+      throw new Error(
+        `Moralis API error: ${response.status} ${response.statusText}`
+      );
+    }
+
+    const asJson = await response.json();
+    if (!isMoralisBalanceResponse(asJson)) {
+      throw new Error('Invalid Moralis response');
+    }
```

Comment:
Nitpick: wrap `.json()` call as it can fail? Maybe not needed, just thinking out loud.